_drafts/SC1-4-3-text-bg-contrast.md

```diff
@@ -0,0 +1,198 @@
+- [F83: Failure of Success Criterion 1.4.3 and 1.4.6 due to using background images that do not provide sufficient contrast with foreground text (or images of text)](https://www.w3.org/TR/WCAG20-TECHS/F83.html)
+
+## Assumptions
+- Code validates to a published grammar (eg. 4.1.1 and 4.1.2)
+
+## Test properties
+| Properties        | Values
+|-------------------|-----------
+| Test name         | Text-background contrast
+| Success criterion | 1.4.3 Contrast minimum
+| Test mode         | semiautomated
+| Test environment  | Rendered page
+| Test Subject      | Single page
+| User profile      | Requires sight
+
+## Test procedure
+
+<!---
+Contrast of links to text and visited links etc is a separate criteria.
+
+For now this ruleset does not cover text in images, except SVG, or text in canvas or video elements or WebGL or Flash etc.
+
+Not sure about the order of the steps as there is no one situation that would pass without checking the others. Wonder if this would mean changing how things are grouped into:
+1. determine text stuff.
+2. determine ratio required.
+3. determine what to contrast it with (and if test can be automated).
+4. check if requirement is met.
+
+ - gradient text??  text masks??  filters on text/bg rendered or not??
+ - not aliased text for pixel to pixel comparison with background image, return percentage of checks that pass
+ - what about canvas or video behind text??
+ - Should this also apply to PDF??
+-->
+
+
+### Selector
+Test method: [automatic]
+
+Find each text node within the page.
+eg. node.nodeType = 3;
+eg. //\*[text()]
+
+
+For each text node:
+Step 1
+Does text node have a foreground color set if yes, step 2
+Step 2
+Does te a background color set
+
+
+### Step 1
+Test method: [automatic][AUTO]
+
+Check if the text node has an applied color property (ie. not default).
+
+if no, continue with [step 2](#step-2)
+
+else, continue with [step 3](#step-3)
+
+
+### Step 2
+Test method: [automatic][AUTO]
+
+Check if there is a background applied behind the text (ie. not default).
+
+if yes, return a fail
+
+| Outcome  | Failed
+|----------|-----
+| ID       | {{ page.rule_id }}-fail2
+| Error    | Text node has no color property, but there is a background set behind it.
+
+else, return a pass
+
+| Outcome  | Pass
+|----------|-----
+| ID       | {{ page.rule_id }}-pass2
+
+
+### Step 3
+Test method: [automatic][AUTO]
+
+Check if there is a background color applied behind the text (ie. not default). (Note: which element?)
+
+if yes, return a pass
+
+| Outcome  | Pass
+|----------|-----
+| ID       | {{ page.rule_id }}-pass3
+
+else, return a fail
+
+| Outcome  | Failed
+|----------|-----
+| ID       | {{ page.rule_id }}-fail3
+| Error    | Text node has color property, but there is no background colour set behind it.
+
+---
+
+### Scenario 1 - text and ancestor background colors
+Test method: [automatic]
```

Comment:
My draft ruleset had this brain dump in steps. We renamed them scenarios as a temporary measure. The intention is to replace them with fully though out steps.
We don't expect this pull request to be accepted, it was more for visibility/comments. However, we could move the scenarios into a markdown comment, if that helps.